src/geometry/algorithms/ArborX_Distance.hpp

```diff
@@ -198,6 +200,50 @@ struct distance<PointTag, TriangleTag, Point, Triangle>
   }
 };
 
+// distance point-tetrahedron
+template <typename Point, typename Tetrahedron>
+struct distance<PointTag, TetrahedronTag, Point, Tetrahedron>
+{
+  static constexpr int DIM = dimension_v<Point>;
+  using Coordinate = coordinate_type_t<Tetrahedron>;
+
+  KOKKOS_FUNCTION static auto apply(Point const &point, Tetrahedron const &tet)
+  {
+    static_assert(GeometryTraits::dimension_v<Point> == 3);
+
+    constexpr int N = 4;
+    Kokkos::Array<decltype(tet.a), N> v = {tet.a, tet.b, tet.c, tet.d};
+
+    // For every plane check that the vertex lies within the same halfspace as
+    // the other tetrahedron vertex (similar to the current intersects
+    // algorithm). If not, compute the distance to the corresponding triangle.
```

Comment:
What if it lies in multiple half spaces? For the triangle-point intersection we are looking at multiple zones defined by rays orthogonal to the edges (in the triangle's plane), not the halfspaces. Shouldn't we do the same here or just take the minimum of the distances from the 4 triangles that define the tetrahedron?

Reply:
You are right.